src/main/ruby/core/io.rb

```diff
@@ -821,7 +822,21 @@ def self.pipe(external=nil, internal=nil, options=nil)
     lhs.send :initialize_allocated
     rhs.send :initialize_allocated
 
-    Truffle.invoke_primitive :io_connect_pipe, lhs, rhs
+    fds = []
+    FFI::MemoryPointer.new(:int, 2) do |ptr|
+      r = Truffle::POSIX.pipe(ptr)
+      Errno.handle if r == -1
+      fds = ptr.read_array_of_int(2)
+    end
```

Comment:
Style suggestion: could this be the following?
```
fds = FFI::MemoryPointer.new(:int, 2) do |ptr|
 ...
 ptr.read_array_of_int(2)
end
```

Reply:
This only works if `.new` is overriden as otherwise `new` always returns the new instance.
But this is the case for us it turns out.
On the other hand, it seems the standard FFI gem returns the new instance in this case.
It seems safer to me to not use this hidden and slightly incompatible feature.
`break ptr.read_array_of_int(2)` would work but it's not so nice.